packages/statemanager/test/stateManager.spec.ts

```diff
@@ -281,4 +284,36 @@ describe('StateManager -> General', () => {
     zeroAccount = await newPartialStateManager2.getAccount(Address.zero())
     assert.ok(zeroAccount!.nonce === zeroAddressNonce)
   })
+  it.skipIf(isBrowser() === true)(
+    'should create a statemanager fromProof with opts preserved',
+    async () => {
+      const trie = await Trie.create({ useKeyHashing: false })
+      const sm = new DefaultStateManager({ trie })
+      const pk = hexToBytes('0x9f12aab647a25a81f821a5a0beec3330cd057b2346af4fb09d7a807e896701ea')
+      const address = Address.fromPrivateKey(pk)
+      const account = new Account()
+      await sm.putAccount(address, account)
+      await sm.putContractStorage(address, setLengthLeft(intToBytes(0), 32), intToBytes(32))
+      const storage = await sm.dumpStorage(address)
+      const keys = Object.keys(storage)
+      const proof = await sm.getProof(
+        address,
+        keys.map((key) => hexToBytes(key))
+      )
+      const newTrie = await Trie.createFromProof(
+        proof.accountProof.map((e) => hexToBytes(e)),
+        { useKeyHashing: false }
+      )
+      const partialSM = await DefaultStateManager.fromProof(proof, true, { trie: newTrie })
```

Comment:
I see here, but we already instantiated the `newTrie` with the proof, right? So does this actually add anything? (Except that it checks that we indeed keep the `useKeyHashing` false?). Maybe for completeness add another key/value pair here (a `proof2`) and insert this in the state manager here. Then one can check for both these key/value pairs that they are correct.

Reply:
It adds the storage values from the proof and the test then verifies that they are returned and match the expected result.  I will add a second account and verify that as well.